### PR TITLE
Updated for CakePHP 4 and PHP 7

### PR DESCRIPTION
**Pull request by mirko-pagliai**
See #10

### CODE REVIEW

**Comment by mirko-pagliai** [comment]
It seems ok.

**Comment by mirko-pagliai** [comment]
But for now, since CakePHP 4 is still in a development phase, it is worthwhile to create a branch instead of publishing a release

File: composer.json [hunk 1]
```diff
@@ -18,12 +19,13 @@
...
-        "cakephp/cakephp": "^3.7",
-        "wyrihaximus/html-compress": "^2.0 || ^1.0"
+        "php": ">=5.5.9",
+        "cakephp/cakephp": "^3.7 || 4.x-dev as 4.0.0",
+        "cakephp/chronos": "^1.2 || 2.x-dev as 2.0.0",
+        "wyrihaximus/html-compress": "^1.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^5.7",
+        "phpunit/phpunit": "^5.7|^6.0|^8.0",
```
Comment (WyriHaximus) [inline composer.json]: Makes sense, but why are you skipping 7?

File: composer.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
         "compress",
         "minify",
         "cakephp",
-        "cakephp3"
+        "cakephp3",
+        "cakephp4"
```
Comment (WyriHaximus) [inline composer.json]: 👍

File: .gitignore [hunk 3]
```diff
@@ -1 +1,3 @@
+.phpunit.result.cache
+composer.lock
```
Comment (WyriHaximus) [inline .gitignore]: Could you revert this, the lock file is used to ensure this plugin works with the current, lowest, and latest dependencies. Plus it's best practice to commit so who ever picks it up to work on has a set of working dependencies locked and ready to use.

File: composer.json [hunk 4]
```diff
@@ -18,12 +19,13 @@
         }
     ],
     "require": {
-        "php": "^7.0 || ^5.6",
-        "cakephp/cakephp": "^3.7",
-        "wyrihaximus/html-compress": "^2.0 || ^1.0"
+        "php": ">=5.5.9",
+        "cakephp/cakephp": "^3.7 || 4.x-dev as 4.0.0",
+        "cakephp/chronos": "^1.2 || 2.x-dev as 2.0.0",
+        "wyrihaximus/html-compress": "^1.0"
```
Comment (WyriHaximus) [inline composer.json]: Could you also include html-compress `^2.0` here, released it yesterday 🎉 .

File: composer.json [hunk 5]
```diff
@@ -18,12 +19,13 @@
         }
     ],
     "require": {
-        "php": "^7.0 || ^5.6",
-        "cakephp/cakephp": "^3.7",
-        "wyrihaximus/html-compress": "^2.0 || ^1.0"
+        "php": ">=5.5.9",
+        "cakephp/cakephp": "^3.7 || 4.x-dev as 4.0.0",
+        "cakephp/chronos": "^1.2 || 2.x-dev as 2.0.0",
```
Comment (WyriHaximus) [inline composer.json]: 👍

File: composer.json [hunk 6]
```diff
@@ -18,12 +19,13 @@
         }
     ],
     "require": {
-        "php": "^7.0 || ^5.6",
-        "cakephp/cakephp": "^3.7",
-        "wyrihaximus/html-compress": "^2.0 || ^1.0"
+        "php": ">=5.5.9",
```
Comment (WyriHaximus) [inline composer.json]: Can you also revert this, the current version targeting also allows PHP8, 9, etc. And we don't know yet if those are compatible. Rather lock them down to 5 and 7 until this is tested on the next version and then release a new version with the changes required for 8.

File: composer.json [hunk 7]
```diff
@@ -37,11 +39,5 @@
         "psr-4": {
             "WyriHaximus\\CakePHP\\Tests\\MinifyHtml\\View\\Helper\\": "tests/"
         }
-    },
-    "config": {
-        "sort-packages": true,
-        "platform": {
-            "php": "5.6"
-        }
```
Comment (WyriHaximus) [inline composer.json]: Can you revert this, this is in to ensure everything we require works with the lowest supported PHP version.

**Comment by WyriHaximus** [comment]
> But for now, since CakePHP 4 is still in a development phase, it is worthwhile to create a branch instead of publishing a release

`master` is fine, won't release a new version until 4 is out anyway 😄

File: composer.json (hunk 1 above)
Comment (mirko-pagliai) [reply composer.json]: If I'm not mistaken, CakePHP 3 is not compatible with phpunit 7. Instead, CakePHP 4 requires at least phpunit 8.

File: .gitignore (hunk 3 above)
Comment (mirko-pagliai) [reply .gitignore]: Ok

**Comment by mirko-pagliai** [comment]
Now it should be ok. Let me know.

**Comment by mirko-pagliai** [comment]
I'm afraid this happens because of the lock file. It is usually not a good thing to include it

**Comment by mirko-pagliai** [comment]
@WyriHaximus ?

**Comment by WyriHaximus** [comment]
@mirko-pagliai will get back to this later today or tomorrow. Finishing up some things for another project first

File: composer.json (hunk 1 above)
Comment (WyriHaximus) [reply composer.json]: Yeah you're right 3 targets 5 and 6 while 4 targets 8 :+1:

**Comment by WyriHaximus** [comment]
LGTM me thanks :+1: I'll fix the failing builds

**Comment by WyriHaximus** [comment]
Fixed the build: https://travis-ci.com/WyriHaximus/MinifyHtml/builds/118884420

Can you confirm it works in your CakePHP 4 project when you target `dev-master`?

**Comment by mirko-pagliai** [comment]
hi, it seems to me that it works

**Comment by WyriHaximus** [comment]
:tada:

**Pull request merged**